main/tasks/stratum_task.c

```diff
@@ -48,313 +44,335 @@ struct timeval tcp_rcv_timeout = {
 
 bool is_wifi_connected() {
     wifi_ap_record_t ap_info;
-    if (esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK) {
-        return true;
-    } else {
-        return false;
-    }
-}
-
-void cleanQueue(GlobalState * GLOBAL_STATE) {
-    ESP_LOGI(TAG, "Clean Jobs: clearing queue");
-    GLOBAL_STATE->abandon_work = 1;
-    queue_clear(&GLOBAL_STATE->stratum_queue);
-
-    pthread_mutex_lock(&GLOBAL_STATE->valid_jobs_lock);
-    ASIC_jobs_queue_clear(&GLOBAL_STATE->ASIC_jobs_queue);
-    for (int i = 0; i < 128; i = i + 4) {
-        GLOBAL_STATE->valid_jobs[i] = 0;
-    }
-    pthread_mutex_unlock(&GLOBAL_STATE->valid_jobs_lock);
+    return esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK;
 }
 
-void stratum_reset_uid(GlobalState * GLOBAL_STATE)
+void stratum_task_primary(void * pvParameters)
 {
-    ESP_LOGI(TAG, "Resetting stratum uid");
-    GLOBAL_STATE->send_uid = 1;
-}
+    GlobalState * GLOBAL_STATE = (GlobalState *) pvParameters;
 
+    StratumConnection * connection = &GLOBAL_STATE->connections[0];
 
-void stratum_close_connection(GlobalState * GLOBAL_STATE)
-{
-    if (GLOBAL_STATE->sock < 0) {
-        ESP_LOGE(TAG, "Socket already shutdown, not shutting down again..");
-        return;
-    }
+    stratum_task_init_connection(connection);
+    connection->id = 0;
+    connection->host = GLOBAL_STATE->SYSTEM_MODULE.pool_url;
+    connection->port = GLOBAL_STATE->SYSTEM_MODULE.pool_port;
+    connection->username = GLOBAL_STATE->SYSTEM_MODULE.pool_user;
+    connection->password = GLOBAL_STATE->SYSTEM_MODULE.pool_pass;
+
+    ESP_LOGI(TAG, "Opening connection to primary pool: %s:%d", connection->host, connection->port);
 
-    ESP_LOGE(TAG, "Shutting down socket and restarting...");
-    shutdown(GLOBAL_STATE->sock, SHUT_RDWR);
-    close(GLOBAL_STATE->sock);
-    cleanQueue(GLOBAL_STATE);
-    vTaskDelay(1000 / portTICK_PERIOD_MS);
+    stratum_task("primary_pool", GLOBAL_STATE, connection);
```

Comment:
Reverse tag (`pool_primary` and `pool_secondary`), for easier grepping.

Also, not sure if the naming is in line with what's used in AxeOS. There it's pool and fallback pool. That might actually be clearer?

Reply:
Good feedback will change

Reply:
```
025-02-19 23:10:59,527 INFO I (1439078) primary_pool: Abandoning the Stratum queues.
2025-02-19 23:10:59,529 INFO I (1439198) secondary_pool: Abandoning the Stratum queues.
2025-02-19 23:11:01,396 INFO I (1441058) primary_pool: rx: {"result":true,"error":null,"id":180}
2025-02-19 23:11:01,396 INFO I (1441058) primary_pool: message result accepted
```

vs.

```
025-02-19 23:10:59,527 INFO I (1439078) pool_primary: Abandoning the Stratum queues.
2025-02-19 23:10:59,529 INFO I (1439198) pool_secondary: Abandoning the Stratum queues.
2025-02-19 23:11:01,396 INFO I (1441058) pool_primary: rx: {"result":true,"error":null,"id":180}
2025-02-19 23:11:01,396 INFO I (1441058) pool_primary: message result accepted
```

Reply:
Actually not sure about this. It kinda feels more natural to me to read the current version of this.